packages/frontend/components/filter/component.tsx

```diff
@@ -1,55 +1,50 @@
-import type { ProductType } from '@vse-bude/shared';
+import type { ProductQuery } from '@vse-bude/shared';
 import { useRouter } from 'next/router';
 import { useAppDispatch, useTypedSelector } from '@hooks';
 import { fetchProducts } from 'store/product';
-import { useEffect, useState } from 'react';
-import { css } from '@emotion/react';
-import { Button } from '@primitives';
-import { PostTypeModal } from '@components/make-a-post/type-of-post';
-import { ProductGrid } from './product-grid/component';
-
-interface RequestOptions {
-  limit?: number;
-  type?: ProductType;
-  category?: string;
-}
+import { useEffect } from 'react';
+import { Routes } from '@enums';
+import { Container } from '@components/primitives';
+import { ProductGrid } from './product-grid';
+import { FilterHeader } from './filter-header';
+import { Pagination } from './pagination';
+import { ProductsLoader } from './products-loader';
 
 export const Filter = () => {
-  const [isOpenTypeOfPost, setIsOpenTypeOfPost] = useState(false);
-  const { query } = useRouter();
-  const { list } = useTypedSelector((store) => store.product);
+  const { query, push } = useRouter();
+  const { list, loading } = useTypedSelector((store) => store.product);
   const dispatch = useAppDispatch();
 
   useEffect(() => {
-    const filter: RequestOptions =
-      query.filter && JSON.parse(query.filter as string);
+    if (!query?.filter) {
+      dispatch(fetchProducts({}));
 
-    dispatch(
-      fetchProducts({
-        limit: filter?.limit,
-        type: filter?.type,
-        categoryId: filter?.category,
-      }),
-    );
-  }, [dispatch, query]);
+      return;
+    }
+    try {
+      const filter: ProductQuery =
+        query.filter && JSON.parse(query.filter as string);
+      dispatch(fetchProducts(filter));
+    } catch (_) {
+      push({
+        pathname: Routes.ITEMS,
+      });
+    }
```

Comment:
1. move to function with returns filters object
2. dispatch(fetchProducts(filter));

in both cases